doc/mutable_references.md

```diff
@@ -12,77 +12,80 @@ a heap (can't dynamically allocate objects). We subvert Rust's systems by using
-meaning you can either have one mutable reference to part of memory or an "infinite" amount of 
+Borrowing is Rust's mechanism to allow references to a part of memory. Similarly to C++ and other
+languages, it makes it possible to pass large structures simply by passing a reference to that structure, 
+rather than being forced to copy over the entire structure.
+However, Rust's compiler limits your borrows so that it doesn't run into the reader-writer problem,
+limiting you to either a single mutable reference to a part of memory or an "infinite" amount of 
 read-only references. Given that read and write are mutally exclusive for borrows, it's impossible
 to run into a race condition using safe rust.
 
-But, what does this mean for Tock? It's a single-threaded enviroment right now, so clearly we won't 
-have race-conditions anyway. A problem arises when the borrowing system of Rust clashes with
+But what does this mean for Tock? As it is currently in a single-threaded enviroment, we clearly will not 
+have race-conditions. A problem arises when the borrowing system of Rust clashes with
 event-driven code without a Heap. 
 
 ## <a href="#issues"></a> Issues with Borrowing in Event-Driven code without a Heap 
-Embedded Devices are often low-resource device -- they don't have resources, such as space to waste.
+Embedded Devices are often low-resource devices -- they don't have resources, such as space to waste.
 Thus it is extremely common to share buffers between drivers, and applications using those drivers
-normally with a pointer to the memory. But, with pointers how can we 'revoke' access when the driver
-is done sharing its buffer? How can we prevent one process from mangaling the data of another?
+normally do so with a pointer to the memory that holds the buffer. However, this raises several potential 
+problems. With pointers, how can we "revoke" access when the driver is done sharing its buffer? How can we prevent one 
+process from mangling the data of another?
 
-We solve this issue of uniquely sharing memory with the memory container abstraction, TakeCell which
+We solve this issue of uniquely sharing memory with the memory container abstraction, TakeCell, which
 will be explained below.
 
-Besides the difficulty of sharing memory safey, it's difficult to have an overall dynamic memory 
-manager for these devices, the differences between the needs of the memory manager would make it 
-such that no one is happy. Furthermore, what would happen if we try to dynamically allocate memory
-in the Kernal or Drivers and memory exhausted... crashing seems like a bad idea.
+Besides the difficulty of sharing memory safely, it's also difficult to have an overall dynamic memory 
+manager for these devices, as the differences between the needs of the memory manager would make it 
+such that no one is happy. Furthermore, if we dynamically allocated memory in the Kernel or for the Drivers and 
```

Comment:
:P prefer: Furthermore, what would happen if we try to dynamically allocate memory
 -in the Kernal or Drivers and memory exhausted... crashing seems like a bad idea.